MonoGame.Framework/Graphics/Texture2D.OpenGL.cs

```diff
@@ -70,73 +70,61 @@ public partial class Texture2D : Texture
         private void PlatformConstruct(int width, int height, bool mipmap, SurfaceFormat format, SurfaceType type, bool shared)
         {
             this.glTarget = TextureTarget.Texture2D;
+            format.GetGLFormat(GraphicsDevice, out glInternalFormat, out glFormat, out glType);
+            // GL texture is generated on first access in PlatformSetData
```

Comment:
Turns out the calls to `glTexImage2D` and `glCompressedTexImage2D` with null data pointers at texture creation are necessary, otherwise `SetData()` fails with `INVALID_VALUE` and it complains about the texture format.